core/src/main/java/io/questdb/cairo/TableWriter.java

```diff
@@ -1554,15 +1554,77 @@ private static ColumnVersionWriter openColumnVersionFile(FilesFacade ff, Path pa
         }
     }
 
-    private static void attachPartitionCheckFilesMatchVarLenColumn(FilesFacade ff, Path path, long partitionSize) {
+    private void attachPartitionCheckFilesMatchFixedColumn(FilesFacade ff, Path path, int columnType, long partitionSize, String columnName, long columnNameTxn) {
+        TableUtils.dFile(path, columnName, columnNameTxn);
+        if (ff.exists(path.$())) {
+            long fileSize = ff.length(path);
+            if (fileSize < partitionSize << ColumnType.pow2SizeOf(columnType)) {
+                throw CairoException.instance(0)
+                        .put("Column file is too small. ")
+                        .put("Partition files inconsistent [file=")
+                        .put(path)
+                        .put(", expectedSize=")
+                        .put(partitionSize << ColumnType.pow2SizeOf(columnType))
+                        .put(", actual=")
+                        .put(fileSize)
+                        .put(']');
+            }
+            return;
+        }
+        throw CairoException.instance(0).put("Column file does not exist [path=").put(path).put(']');
+    }
+
+    private void attachPartitionCheckFilesMatchMetadata(FilesFacade ff, Path path, RecordMetadata metadata, long partitionSize) throws CairoException {
+        // for each column, check that file exists in the partition folder
+        int rootLen = path.length();
+        for (int columnIndex = 0, size = metadata.getColumnCount(); columnIndex < size; columnIndex++) {
+            try {
+                int columnType = metadata.getColumnType(columnIndex);
+                final String columnName = metadata.getColumnName(columnIndex);
+                long columnNameTxn = columnVersionWriter.getDefaultColumnNameTxn(columnIndex);
+
+                switch (ColumnType.tagOf(columnType)) {
+                    case ColumnType.INT:
+                    case ColumnType.LONG:
+                    case ColumnType.BOOLEAN:
+                    case ColumnType.BYTE:
+                    case ColumnType.TIMESTAMP:
+                    case ColumnType.DATE:
+                    case ColumnType.DOUBLE:
+                    case ColumnType.CHAR:
+                    case ColumnType.SHORT:
+                    case ColumnType.FLOAT:
+                    case ColumnType.LONG256:
+                        // Consider Symbols as fixed, check data file size
```

Comment:
nit: why do we have this comment here? This line and what's above seems to be unrelated with symbols.